src/ethernet.h

```diff
@@ -65,17 +65,17 @@ struct ether_header {
 	uint8_t ether_dhost[ETH_ALEN];
 	uint8_t ether_shost[ETH_ALEN];
 	uint16_t ether_type;
-} __attribute__((__gcc_struct__)) __attribute((__packed__));
+};
 #endif
 
 #ifndef HAVE_STRUCT_ARPHDR
-struct arphdr {
+PACKED(struct arphdr {
```

Comment:
Still some `PACKED`s? I think they can *all* be removed.

Reply:
Ah, sorry, you're right. [OpenBSD][one] still [uses][two] them [for some][thr] structs, but it doesn't seem necessary:

[one]: https://github.com/openbsd/src/blob/2207c4325726fdc5c4bcd0011af0fdf7d3dab137/sys/netinet/ip6.h#L72
[two]: https://github.com/openbsd/src/blob/2207c4325726fdc5c4bcd0011af0fdf7d3dab137/sys/netinet/icmp6.h#L70
[thr]: https://github.com/openbsd/src/blob/2207c4325726fdc5c4bcd0011af0fdf7d3dab137/sys/netinet/icmp6.h#L212

https://github.com/NetBSD/src/commit/427831ba4bdf388aecf3a378de8faf3a4d44a462


I rearranged lines into their original position, the diff is smaller than it looks:

    $ git diff upstream/1.1..HEAD src/{ipv4,ipv6,ethernet}.h

```diff
diff --git a/src/ethernet.h b/src/ethernet.h
index 3b074b7f..4ef9dab9 100644
--- a/src/ethernet.h
+++ b/src/ethernet.h
@@ -65,7 +65,7 @@ struct ether_header {
 	uint8_t ether_dhost[ETH_ALEN];
 	uint8_t ether_shost[ETH_ALEN];
 	uint16_t ether_type;
-} __attribute__((__gcc_struct__)) __attribute((__packed__));
+};
 #endif

 #ifndef HAVE_STRUCT_ARPHDR
@@ -75,7 +75,7 @@ struct arphdr {
 	uint8_t ar_hln;
 	uint8_t ar_pln;
 	uint16_t ar_op;
-} __attribute__((__gcc_struct__)) __attribute((__packed__));
+};

 #define ARPOP_REQUEST 1
 #define ARPOP_REPLY 2
@@ -93,7 +93,7 @@ struct  ether_arp {
 	uint8_t arp_spa[4];
 	uint8_t arp_tha[ETH_ALEN];
 	uint8_t arp_tpa[4];
-} __attribute__((__gcc_struct__)) __attribute((__packed__));
+};
 #define arp_hrd ea_hdr.ar_hrd
 #define arp_pro ea_hdr.ar_pro
 #define arp_hln ea_hdr.ar_hln
diff --git a/src/ipv4.h b/src/ipv4.h
index 708e5c1f..a412c605 100644
--- a/src/ipv4.h
+++ b/src/ipv4.h
@@ -83,7 +83,7 @@ struct ip {
 	uint8_t ip_p;
 	uint16_t ip_sum;
 	struct in_addr ip_src, ip_dst;
-} __attribute__((__gcc_struct__)) __attribute((__packed__));
+};
 #endif

 #ifndef IP_OFFMASK
@@ -145,7 +145,7 @@ struct icmp {
 #define icmp_radv icmp_dun.id_radv
 #define icmp_mask icmp_dun.id_mask
 #define icmp_data icmp_dun.id_data
-} __attribute__((__gcc_struct__)) __attribute((__packed__));
+};
 #endif

 #endif
diff --git a/src/ipv6.h b/src/ipv6.h
index cfa6f239..7c06de3d 100644
--- a/src/ipv6.h
+++ b/src/ipv6.h
@@ -51,7 +51,7 @@ struct ip6_hdr {
 	} ip6_ctlun;
 	struct in6_addr ip6_src;
 	struct in6_addr ip6_dst;
-} __attribute__((__gcc_struct__)) __attribute((__packed__));
+};
 #define ip6_vfc ip6_ctlun.ip6_un2_vfc
 #define ip6_flow ip6_ctlun.ip6_un1.ip6_un1_flow
 #define ip6_plen ip6_ctlun.ip6_un1.ip6_un1_plen
@@ -70,7 +70,7 @@ struct icmp6_hdr {
 		uint16_t icmp6_un_data16[2];
 		uint8_t icmp6_un_data8[4];
 	} icmp6_dataun;
-} __attribute__((__gcc_struct__)) __attribute((__packed__));
+};
 #define ICMP6_DST_UNREACH_NOROUTE 0
 #define ICMP6_DST_UNREACH 1
 #define ICMP6_PACKET_TOO_BIG 2
@@ -90,7 +90,7 @@ struct icmp6_hdr {
 struct nd_neighbor_solicit {
 	struct icmp6_hdr nd_ns_hdr;
 	struct in6_addr nd_ns_target;
-} __attribute__((__gcc_struct__)) __attribute((__packed__));
+};
 #define ND_OPT_SOURCE_LINKADDR 1
 #define ND_OPT_TARGET_LINKADDR 2
 #define nd_ns_type nd_ns_hdr.icmp6_type
@@ -103,7 +103,7 @@ struct nd_neighbor_solicit {
 struct nd_opt_hdr {
 	uint8_t nd_opt_type;
 	uint8_t nd_opt_len;
-} __attribute__((__gcc_struct__)) __attribute((__packed__));
+};
 #endif

 #endif
```